device/src/main/java/com/softserve/lv460/device/service/impl/DeviceDataServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.softserve.lv460.device.service.impl;
+
+import com.softserve.lv460.device.config.DeviceCacheConfig;
+import com.softserve.lv460.device.config.PropertiesConfig;
+import com.softserve.lv460.device.document.DeviceData;
+import com.softserve.lv460.device.exceptions.DeviceNotRegisteredException;
+import com.softserve.lv460.device.exceptions.Massages;
+import com.softserve.lv460.device.repositiry.DeviceDataRepository;
+import com.softserve.lv460.device.service.DeviceDataService;
+import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@Service
+@AllArgsConstructor
+public class DeviceDataServiceImpl implements DeviceDataService {
+  private List<DeviceData> batch = Collections.synchronizedList(new ArrayList<>());
```

Comment:
Let's review this  Collections.synchronizedList later, but for now, i guess it will be more efficient to use LinkedList instead of ArrayList as a backed collection, since you're mostly adding elements to collection and LinkedList has constant time complexity for that